src/main/java/com/csc530/sat/type/DDType.java

```diff
@@ -0,0 +1,15 @@
+package com.csc530.sat.type;
+
+/**
+ * A type that the decision diagram can reason about
+ *
+ * @param <T>
+ *            the java type that this type describes
+ */
+public interface DDType<T> {
```

Comment:
Object is more complicated, It can't be any old object it a specific type that the Decision Diagram can reason about